sysl2/sysl/exprOp.go

```diff
@@ -33,6 +33,10 @@ func neInt64(lhs, rhs *sysl.Value) *sysl.Value {
 	return MakeValueBool(lhs.GetI() != rhs.GetI())
 }
 
+func neString(lhs, rhs *sysl.Value) *sysl.Value {
+	return MakeValueBool(lhs.GetS() != rhs.GetS())
+}
```

Comment:
Suggestion (easier to show than explain):
```go
func not(f func(lhs, rhs *sysl.Value) *sysl.Value) func(lhs, rhs *sysl.Value) *sysl.Value {
	return func(lhs, rhs *sysl.Value) *sysl.Value {
		return MakeValueBool(!(f(lhs, rhs)))
	}
}
```
With this in place, use `not(cmpString)` in lieu of writing and using `neString`, and likewise with other negations (`not(cmpInt)` vs `neInt64`, `not(gtInt64)` vs `leInt64`, …).